redis/config.go

```diff
@@ -55,5 +56,9 @@ func (r *Config) Validate() error {
 		return errors.New("Redis host missing")
 	}
 
+	if r.Username != "" && r.Password == "" {
+		return errors.New("password must be set, if username is provided")
```

Comment:
Sorry, but after testing this out, the validation error on its own doesn't seem clear enough as to which password it refers to, so I would prefix it with a *lowered* `redis`.

Reply:
With _Redis_ being a proper name, I would not put it to lowercase. Furthermore, it is written in an uppercase error just three lines above. I am going to prefix it now with "Redis", unless someone else wants to philosophize more about an error message.